### ISSUE TITLE
Sudden OOM error with Tensorflow after several successfully trained batches

### ISSUE THREAD

**Issue by JotaRodriguez94** [tensorflow/tensorflow#36736]
**System information** 
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution: Ubuntu 18.04.3 LTS 
- TensorFlow installed from (source or binary): tensorflow installed from docker image
- TensorFlow version (use command below): v2.1.0
- Python version: 3.6.9
- GCC/Compiler version (if compiling from source): 8.3.0
- CUDA/cuDNN version: V10.1.243
- GPU model and memory: NVIDIA Corporation GM200 [GeForce GTX TITAN X] [10de:1132] and 12 GB of memory

**Description**: I'm running an Encoder-Decoder architecture to do video captioning. For this I created three classes based on keras.Model. The first model has several 3D convolutional layers, normalization and 3D max pooling layers and I use it to describe the video, it is initialized and called inside the Encoder model. This encoder model is called once for each video and passes information to the decoder model which is called several times (as many as the number of words in the caption).

**Input shape**: The dimensions of the input video are (128,224,224,3) where 128 indicates the number of frames and the dimensions of the sentences are (1,55) where 55 indicates the number of tokens.

**Problem**: I'm using a batch of 1 (video) and the model trains well up to a certain number of videos and then throws an OOM error as seen in the picture. I have tried to lower the number of frames to 64 and 32 but the same error keeps coming up but, interestingly, the error takes longer to come up as I lower the frames, this makes me think that in a certain place the memory is being used in a cumulative way. From the error I can deduce that it appears in the backpropagation of the first layer maxpoling3d. However, I don't understand why the model runs well for some videos and then throws the error, if it was by memory, it should break in the first video, right? I attach the training function I'm using and the data generator.

![Selección_002](https://user-images.githubusercontent.com/18648306/74474042-b057be00-4e72-11ea-8de4-6cd0486ac4c0.png)
![Selección_003](https://user-images.githubusercontent.com/18648306/74474072-bcdc1680-4e72-11ea-961c-dd4532beda0c.png)
![Selección_001](https://user-images.githubusercontent.com/18648306/74473583-d7fa5680-4e71-11ea-8c86-f8f7420cd139.png)

**Comment by gadagashwini** [comment]
@JotaRodriguez94, Thanks for reporting this issue. 
Could you provide the complete standalone code to replicate the reported issue. Thanks!

**Comment by gadagashwini** [comment]
@JotaRodriguez94, Can you share the complete code to analyze the issue. Thanks

**Comment by gadagashwini** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by i4never** [comment]
Check batch length in your training set may help. In my situation (nlp task), batch data has dynamic sequence length (which depends on the longest sequence within that batch). So chances are that the records in that OOM batch may contain very long sequence.
